### PR TITLE
[W09H02] test(CleanDatasetTest, DelayComparedToTravelTimeTest): Add template and first tests

### PR DESCRIPTION
**Pull request by hmelder**
Adapted from this pull request https://github.com/MaximilianAnzinger/pgdp2223-tests/pull/175.

- Cleaned up commit history
- Add new Test case for second task
- Rewrite all utility classes 
- Updated JSON blobs

### CODE REVIEW

**Comment by hmelder** [comment]
Added a basic NaN test. This occurs when the departure time equals the arrival time.

Maybe we should manipulated the unmarshalled JSON object and not add new JSON Strings. This would also improve execution time.

**Comment by hmelder** [comment]
> Maybe we should manipulated the unmarshalled JSON object and not add new JSON Strings. This would also improve execution time.

Yea I'm writing another utility class for this

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,6 @@ hs_err_pid*
 
 # macOS files
 .DS_Store
+
+# vim
+*.swp
```
Comment (Anatoly03) [inline .gitignore:33]: Not sure if it’s worth doing for three lines, but perhaps this could be split into another PR?

Comment (hmelder) [reply .gitignore:33]: bruh

Comment (hmelder) [reply .gitignore:33]: Gotta be kidding me. It is separated in the commit history.

Comment (hmelder) [reply .gitignore:33]: > Not sure if it’s worth doing for three lines, but perhaps this could be split into another PR?

What three lines? Have you looked at the changes?

**Comment by nilsreichardt** [comment]
@julian-hartl Could you review this PR because it would replace your PR (#175)?

**Review by MaximilianAnzinger — APPROVED** [review]

File: w09h02/test/pgdp/trains/processing/utils/JSONParser.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  private static final TrainConnection str29CanceledParsed;
+  private static final JSONObject re8BasicDataset;
+  private static final TrainConnection re8BasicParsed;
+
+  static {
+    iceBasicDataset = new JSONObject(JSONDataset.iceBasicBlob);
+    str29CanceledDataset = new JSONObject(JSONDataset.str29CanceledBlob);
+    re8BasicDataset = new JSONObject(JSONDataset.re8BasicBlob);
+
+    iceBasicParsed = TrainConnection.parseFromJSON(iceBasicDataset);
```
Comment (julian-hartl) [inline w09h02/test/pgdp/trains/processing/utils/JSONParser.java:20]: Could you replace the repetitive json parse method with the given static method?

```suggestion
    iceBasicParsed = getTrainConnectionFromJSON(iceBasicDataset);
```

File: w09h02/test/pgdp/trains/processing/utils/JSONParser.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    str29CanceledParsed = TrainConnection.parseFromJSON(str29CanceledDataset);
```
Comment (julian-hartl) [inline w09h02/test/pgdp/trains/processing/utils/JSONParser.java:21]: Same for the other parsings.

File: w09h02/test/pgdp/trains/processing/utils/JSONParser.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    re8BasicParsed = TrainConnection.parseFromJSON(re8BasicDataset);
+  }
+
+  public static TrainConnection getTrainConnectionFromJSON(JSONObject obj) {
```
Comment (julian-hartl) [inline w09h02/test/pgdp/trains/processing/utils/JSONParser.java:25]: This could be private, I guess (does not really matter though).

File: .gitignore (hunk 1 above)
Comment (felixhauptmann) [reply .gitignore:33]: > bruh. It is separated in the commit history.

the commit has to be in a different branchfor it to not get added to the existing PR

Comment (hmelder) [reply .gitignore:33]: > > bruh. It is separated in the commit history.
> 
> the commit has to be in a different branch for it to not get added to the existing PR

I know.

File: w09h02/test/pgdp/trains/processing/utils/JSONParser.java (hunk 2 above)
Comment (hmelder) [reply w09h02/test/pgdp/trains/processing/utils/JSONParser.java:20]: Then you would have another layer of indirection. I created the shell method specifically for converting when running unit tests.

File: w09h02/test/pgdp/trains/processing/utils/JSONParser.java (hunk 4 above)
Comment (hmelder) [reply w09h02/test/pgdp/trains/processing/utils/JSONParser.java:25]: Look at w09h02/test/pgdp/trains/processing/DelayComparedToTravelTimeTest.java

File: .gitignore (hunk 1 above)
Comment (felixhauptmann) [reply .gitignore:33]: > > > bruh. It is separated in the commit history.
> > 
> > 
> > the commit has to be in a different branch for it to not get added to the existing PR
> 
> I know.

It isn't though: https://github.com/hmelder/pgdp2223-tests/commit/18273adb6f53723196e05963d3be243dc26ebd2c

that's why it is linked to this PR.

Comment (hmelder) [reply .gitignore:33]: > > > > bruh. It is separated in the commit history.
> > > 
> > > 
> > > the commit has to be in a different branch for it to not get added to the existing PR
> > 
> > 
> > I know.
> 
> It isn't though: [hmelder@18273ad](https://github.com/hmelder/pgdp2223-tests/commit/18273adb6f53723196e05963d3be243dc26ebd2c)
> 
> that's why it is linked to this PR.

I never contemplated that. This commit was intentionally added to this branch.

Comment (hmelder) [reply .gitignore:33]: It is not in the same commit as the w09h02 changes. You could squash it differently if you want to, but what do you gain?

Comment (nilsreichardt) [reply .gitignore:33]: I have created a separate PR #197 to follow the pattern of small and separated PRs 👍

File: w09h02/test/pgdp/trains/processing/DelayComparedToTravelTimeTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    return parsed;
+  }
+
+  @Test
+  void shouldHandleNaN() {
+    final JSONObject manipulated = DelayComparedToTravelTimeTest.manipulateSTR29();
+
+    final Map<String, Double> result = DataProcessing.delayComparedToTotalTravelTimeByTransport(Stream.of(
+        JSONParser.getTrainConnectionFromJSON(manipulated)
```
Comment (hmelder) [inline w09h02/test/pgdp/trains/processing/DelayComparedToTravelTimeTest.java:35]: @julian-hartl I am using the JSONParser::getTrainConnectionFromJSON here. So it has to be public.

**Comment by hmelder** [comment]
What is the state of the PR?

**Comment by hmelder** [comment]
@MaximilianAnzinger you can merge this now
